src/cli.rs

```diff
@@ -1276,6 +1377,58 @@ impl<'a> BirliContext<'a> {
         Ok(result)
     }
 
+    /// Call `run()` for every channel range in `self.channel_range_sel`.
+    pub fn run_ranges(self) -> Vec<Result<HashMap<String, Duration>, BirliError>> {
+        let ranges = self.channel_range_sel.ranges.clone();
+        let mut ranged_context: BirliContext = BirliContext {
+            corr_ctx: self.corr_ctx,
+            prep_ctx: self.prep_ctx,
+            vis_sel: self.vis_sel,
+            flag_ctx: self.flag_ctx,
+            io_ctx: self.io_ctx,
+            avg_time: self.avg_time,
+            avg_freq: self.avg_freq,
+            num_timesteps_per_chunk: self.num_timesteps_per_chunk,
+            ignore_dut1: self.ignore_dut1,
+            channel_range_sel: self.channel_range_sel,
+        };
+        let original_io_ctx = ranged_context.io_ctx.clone();
+        ranged_context.vis_sel = ranged_context.vis_sel.clone();
+        ranged_context.io_ctx = ranged_context.io_ctx.clone();
```

Comment:
Not your fault, but the fact that cloning is needed here is indicative of a design mismatch. Chatting with Dev, it may be better to pass in a IO context, vis selection and filename to the `run` function on line 1427.